integrations/operator/sidecar/tbot.go

```diff
@@ -139,44 +139,48 @@ func (b *Bot) buildClient(ctx context.Context) (*SyncClient, error) {
 // GetSyncClient gets a client authenticated as the bot. To avoid rebuilding a
 // client for each call, this function caches the client and creates a new one
 // only when the client certs changed (tbot renewed them).
-func (b *Bot) GetSyncClient(ctx context.Context) (*SyncClient, error) {
+func (b *Bot) GetSyncClient(ctx context.Context) (*SyncClient, func(), error) {
 	b.mutex.Lock()
 	defer b.mutex.Unlock()
 
 	if !b.running {
-		return nil, trace.Errorf("bot not started yet")
+		return nil, nil, trace.Errorf("bot not started yet")
 	}
 	// If the bot has not joined the cluster yet or not generated client certs we bail out
 	// This is either temporary or the bot is dead and the manager will shut down everything.
 	storageDestination := b.cfg.Storage.Destination
 	if botCert, err := storageDestination.Read(ctx, identity.TLSCertKey); err != nil || len(botCert) == 0 {
-		return nil, trace.Retry(err, "bot cert not yet present")
+		return nil, nil, trace.Retry(err, "bot cert not yet present")
 	}
 
 	cert, err := b.cfg.Outputs[0].GetDestination().Read(ctx, identity.TLSCertKey)
 	if err != nil || len(cert) == 0 {
-		return nil, trace.Retry(err, "cert not yet present")
+		return nil, nil, trace.Retry(err, "cert not yet present")
 	}
 
 	// This is where caching happens. We don't know when tbot renews the certificates, so we need to check
 	// if the current certificate stored in memory changed since last time. If it did not and we already built a
 	// working client, then we hit the cache. Else we build a new client, replace the cached client with the new one,
 	// and fire a separate goroutine to close the previous client.
 	if b.cachedClient != nil && bytes.Equal(cert, b.cachedCert) {
-		return b.cachedClient, nil
+		return b.cachedClient, b.cachedClient.LockClient(), nil
```

Comment:
A saner approach would be using a `sync.WaitGroup` to count the number of handlers using the client since it's exactly what you need here.
Using mutex to perform this work seems odd.

Reply:
Why saner? AFAIU an RW mutex is essentially a waitgroup for readers with a boolean API that answers "is someone still accessing this or not?"

Reply:
Done in d4bed68facbaa34596699abd469afbd3b7ba8cd3

Reply:
My point about saner is because it's trivial to call Lock instead of RLock which has huge consequences on the behavior of this feature while sync.WaitGroup doesn't allow you to incorrectly block because you mistyped. Sure you can block if you forget to call wg.Done() but it's simpler to detect